amc.groovy

```diff
@@ -24,7 +24,7 @@ def clean     = tryQuietly{ clean.toBoolean() }
 def exec      = tryQuietly{ exec.toString() }
 
 // array of kodi/plex/emby hosts
-def kodi = tryQuietly{ any{kodi}{xbmc}.split(/[ ,|]+/)*.split(/:/).collect{ it.length >= 2 ? [host: it[0], port: it[1] as int] : [host: it[0], port: 8080] } }
+def kodi = tryQuietly{ any{kodi}{xbmc}.split(/[ ,|]+/)*.split(/:[0-9]+/).collect{ it.length >= 2 ? [host: it[0], port: it[1] as int] : [host: it[0], port: 8080] } }
```

Comment:
If the password is a number, it'll split by that. Also, your pattern will split and remove the number part. Please make sure that your split regex only matches `:` and use regex look-ahead to make sure that the port number and `$` follow.